x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/PreAnalyzer.java

```diff
@@ -53,11 +57,36 @@ protected PreAnalysis doPreAnalyze(LogicalPlan plan) {
             }
         });
 
+        List<Enrich> unresolvedEnriches = new ArrayList<>();
         plan.forEachUp(Enrich.class, unresolvedEnriches::add);
 
+        Holder<Boolean> supportsAggregateMetricDouble = new Holder<>(false);
+        Holder<Boolean> supportsDenseVector = new Holder<>(false);
+        plan.forEachDown(p -> p.forEachExpression(UnresolvedFunction.class, fn -> {
+            if (fn.name().equalsIgnoreCase("knn")
+                || fn.name().equalsIgnoreCase("v_cosine")
+                || fn.name().equalsIgnoreCase("v_hamming")
+                || fn.name().equalsIgnoreCase("v_l1_norm")
+                || fn.name().equalsIgnoreCase("v_l2_norm")
+                || fn.name().equalsIgnoreCase("v_dot_product")
+                || fn.name().equalsIgnoreCase("v_magnitude")) {
```

Comment:
Should we add `to_dense_vector`, for consistency's sake?

Reply:
Sure